doc/articles/migrating-to-single-project.md

```diff
@@ -0,0 +1,118 @@
+
+## Upgrading an existing project to Single Project
+
+The following sections detail how to upgrade an existing project to use a single project with the Uno.Sdk. Those modifications are made in-place and **it is strongly recommended to work on a source-controlled environment**.
+
+While following this guide, you can compare with a new empty project created using [our Wizard](xref:Uno.GetStarted.Wizard), or using [dotnet new templates](xref:Uno.GetStarted.dotnet-new).
+
+### Migrating to a Single Project
+
+Migrating from a 5.1 or earlier template is a matter merging all the csproj files for all the platforms heads down to one csproj file.
+
+We'll take the example of an Uno Platform 5.0 blank app, called `MyApp`, to migrate to Single Project :
+1. In a separate folder from your existing app, create a new app using the same name as your app:
+   ```
+   dotnet new unoapp -o MyApp --preset=recommended
+   ```
+   This will allow for taking pieces of the structure with the proper namespaces. If your app was created using the blank template, use `--preset=blank` instead. You can also use the [Live Wizard](xref:Uno.GetStarted.dotnet-new) or the [Visual Studio Wizard](xref:Uno.GettingStarted.CreateAnApp.VS2022) and select the same optional features you used for your app.
+1. Next to the solution, create a `global.json` file
+    ```json
+    {
+        "msbuild-sdks": {
+            "Uno.Sdk": "{Current Version of Uno.WinUI/Uno.Sdk}"
+        }
+    }
+    ```
+1. From the empty app created at step one:
+    - Take the contents of the `MyApp.csproj` and copy it over your own `MyApp.csproj`
+    - Take the folder named `Properties` and copy it over your own `MyApp`
+1. If you have your own `PackageReference` and `ProjectReference`, make sure to preserve them.
+1. From the empty app created at step one, take the `Platforms` folder and copy it over your own `MyApp` folder.
+1. In you app's `MyApp.csproj`, create the following block:
+   ```xml
+   <Choose>
+     <When Condition="'$(TargetFramework)'=='net8.0-ios'">
+     </When>
+     <When Condition="'$(TargetFramework)'=='net8.0-android'">
+     </When>
+     <When Condition="'$(TargetFramework)'=='net8.0-maccatalyst'">
+     </When>
+     <When Condition="'$(TargetFramework)'=='net8.0-windows10.0.19041'">
+     </When>
+     <When Condition="'$(TargetFramework)'=='net8.0-desktop'">
+     </When>
+     <When Condition="'$(TargetFramework)'=='net8.0-browserwasm'">
+     </When>
+   </Choose>
+   ```
+   This will allow to place custom project configuration in each of those sections. Use the ones that are relevant to your project targets.
+1. From the empty app created at step one, take the `App.xaml` and `App.xaml.cs` and copy those over your own `MyApp` folder.
+1. Copy the individual modifications you made to your `AppResources.xaml` and `App.cs` into `App.xaml` and `App.xaml.cs`, respectively. Those can for instance be changes made to `OnLaunched`, `InitializeLogging` or `RegisterRoutes`.
```

Comment:
```suggestion
   This will allow us to place custom project configurations in each of those sections. Use the ones that are relevant to your project targets.
1. From the empty app created at step one, take the `App.xaml` and `App.xaml.cs` and copy those over your own `MyApp` folder.
1. Copy the individual modifications you made to your `AppResources.xaml` and `App.cs` into `App.xaml` and `App.xaml.cs`, respectively. Those can, for instance, be changes made to `OnLaunched`, `InitializeLogging`, or `RegisterRoutes`.
```